src/modules/launcher/Plugins/Microsoft.Plugin.Indexer/SearchHelper/WindowsSearchAPI.cs

```diff
@@ -1,55 +1,52 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Data.OleDb;
+using Microsoft.Plugin.Indexer.Interface;
 using Microsoft.Search.Interop;
 
 namespace Microsoft.Plugin.Indexer.SearchHelper
 {
     public class WindowsSearchAPI
     {
-        public OleDbConnection conn;
-        public OleDbCommand command;
-        public OleDbDataReader WDSResults;
+        public bool DisplayHiddenFiles { get; set; }
+
+        private readonly ISearch WindowsIndexerSearch;
         private readonly object _lock = new object();
-        
+        private readonly UInt32 FILE_ATTRIBUTE_HIDDEN = 0x2;
+
+        public WindowsSearchAPI(ISearch windowsIndexerSearch, bool displayHiddenFiles = false)
+        {
+            this.WindowsIndexerSearch = windowsIndexerSearch;
+            this.DisplayHiddenFiles = displayHiddenFiles;
+        }
 
         public List<SearchResult> ExecuteQuery(ISearchQueryHelper queryHelper, string keyword)
         {
             List<SearchResult> _Result = new List<SearchResult>();
+
             // Generate SQL from our parameters, converting the userQuery from AQS->WHERE clause
             string sqlQuery = queryHelper.GenerateSQLFromUserQuery(keyword);
 
-            // --- Perform the query ---
-            // create an OleDbConnection object which connects to the indexer provider with the windows application
-            using (conn = new OleDbConnection(queryHelper.ConnectionString))
+            // execute the command, which returns the results as an OleDBResults.
+            List<OleDBResult> oleDBResults = WindowsIndexerSearch.Query(queryHelper.ConnectionString, sqlQuery);
+
+            // Loop over all records from the database
+            foreach(OleDBResult oleDBResult in oleDBResults)
             {
-                // open the connection
-                conn.Open();
+                if (oleDBResult.fieldData[0] == DBNull.Value || oleDBResult.fieldData[1] == DBNull.Value || oleDBResult.fieldData[2] == DBNull.Value)
+                    continue;
+                UInt32 fileAttributes = (UInt32)((Int64)oleDBResult.fieldData[2]);
+                bool isFileHidden = (fileAttributes & FILE_ATTRIBUTE_HIDDEN) == FILE_ATTRIBUTE_HIDDEN;
```

Comment:
Is there any reason for type casting it twice and performing a bitwise AND operation? 
The FileAttributes value is set to 2 if the file is hidden. Why not do a simple == comparison?
Ref - https://docs.microsoft.com/en-us/dotnet/api/system.io.fileattributes?view=netcore-3.1

Reply:
Typecasting is required because the object received is Int64. So we first need to convert `object` -> `Int64` and then from `Int64` back to `UInt32`. It's certainly not very neat but I couldn't find any other alternative. Please tell if you might know any other way to do this. 

File Attribute is not 2 when it's file is set to hidden, instead second bit in unsigned representation set to a 1. For example, 0x11 would mean it's read-only and hidden. A direct comparison would not work here. We would need to extract the corresponding bit and check if it is set to 1.

Reply:
Thanks for the clarification about it being a bitwise combination.